crates/router/src/db.rs

```diff
@@ -255,36 +261,75 @@ impl RequestIdStore for KafkaStore {
 impl FraudCheckInterface for KafkaStore {
     async fn insert_fraud_check_response(
         &self,
-        new: FraudCheckNew,
+        new: storage::FraudCheckNew,
     ) -> CustomResult<FraudCheck, StorageError> {
-        self.diesel_store.insert_fraud_check_response(new).await
+        let frm = self.diesel_store.insert_fraud_check_response(new).await?;
+        if let Err(er) = self
+            .kafka_producer
+            .log_fraud_check(&frm, None, self.tenant_id.clone())
+            .await
+        {
+            logger::error!(message = "Failed to log analytics event for payment attempt", error_message = ?er);
+        }
+        Ok(frm)
     }
     async fn update_fraud_check_response_with_attempt_id(
         &self,
-        fraud_check: FraudCheck,
-        fraud_check_update: FraudCheckUpdate,
+        this: FraudCheck,
+        fraud_check: FraudCheckUpdate,
     ) -> CustomResult<FraudCheck, StorageError> {
-        self.diesel_store
-            .update_fraud_check_response_with_attempt_id(fraud_check, fraud_check_update)
+        let frm = self
+            .diesel_store
+            .update_fraud_check_response_with_attempt_id(this, fraud_check)
+            .await?;
+        if let Err(er) = self
+            .kafka_producer
+            .log_fraud_check(&frm, None, self.tenant_id.clone())
             .await
+        {
+            logger::error!(message="Failed to log analytics event for payment attempt {attempt:?}", error_message=?er)
```

Comment:
```suggestion
            logger::error!(message="Failed to log analytics event for fraud check {frm:?}", error_message=?er)
```

Reply:
tbh Idk how does this even compile? since it relies on a variable called attempt which isn't present

Reply:
Done